text/204-golang-bindings.md

```diff
@@ -0,0 +1,669 @@
+
+```ts
+// Typescript
+
+export interface ICluster extends IResource {
+  readonly clusterName: string;
+  readonly clusterArn: string;
+  readonly vpc: ec2.IVpc;
+  readonly connections: ec2.Connections;
+  readonly hasEc2Capacity: boolean;
+  readonly defaultCloudMapNamespace?: cloudmap.INamespace;
+  readonly autoscalingGroup?: autoscaling.IAutoScalingGroup;
+}
+
+export class Cluster extends Resource implements ICluster {
+  public static fromClusterAttributes(scope: Construct, id: string, attrs: ClusterAttributes): ICluster {
+    return new ImportedCluster(scope, id, attrs);
+  }
+  public readonly connections: ec2.Connections = new ec2.Connections();
+  public readonly vpc: ec2.IVpc;
+  public readonly clusterArn: string;
+  public readonly clusterName: string;
+
+  private _defaultCloudMapNamespace?: cloudmap.INamespace;
+  private _hasEc2Capacity: boolean = false;
+  private _autoscalingGroup?: autoscaling.IAutoScalingGroup;
+...
+}
+```
+
+Go struct (derived from TS class) with embedded struct (derived from TS datatype interface):
+
+```go
+type ICluster interface {
+    IResource
+
+    ClusterName()              string
+    ClusterArn()               string
+    Vpc()                      ec2.IVpc
+    Connections()              ec2.Connections
+    HasEc2Capacity()           bool
+    DefaultCloudMapNamespace() cloudmap.INamespace
+    AutoscalingGroup()         autoscaling.IAutoScalingGroup
+}
+
+// Generated interface for Cluster class
+type ClusterIface interface {
+    ICluster
+}
+
+// Generated struct for Cluster class
+type Cluster struct {
+    Resource
+    ICluster
+
+    connections ec2.Connections
+    vpc         ec2.IVpc
+    clusterArn  string
+    clusterName string
+
+    defaultCloudMapNamespace cloudmap.INamespace
+    hasEc2Capacity           bool
+    autoscalingGroup         autoscaling.IAutoScalingGroup
+}
+
+func (c *Cluster) ClusterName() string {
+    return c.clusterName
+}
+
+// ...etc
+```
+
+### Case 4: Abstract Classes
+
+These should be able to be handled much in the same way as regular classes, by generating an interface and a struct. The struct generation is still
+necessary in this case because method implementations can actually be defined on abstract classes in Typescript.
+
+## Other considerations
+
+### Primitive types
+
+#### Date
+
+JSII Date would be satisfied by Go's `time.Time` type.
+
+#### String
+
+JSII String would be satisfied by Go's string type.
+
+#### Number
+
+JSII number does not distinguish the difference between integer and float. This either requires JSII for Go to treat all numbers as float64, or define
+a new jsii.Number type that "wraps" Go number types into the singular JSII Number type.
+
+Typescript numbers are float64 with a max integer size of 2^53 -1. This means that Customers could want to provide a int64 value that would lose
+precision.
+
+#### Any
+
+JSII any maps to Go's empty interface type. JSII for Go should replace all JSII any types with interface{}.
```

Comment:
Will `interface{}` also accept simple types like `int` and `string` and such?

Reply:
What do you mean?